src/map/unit.cpp

```diff
@@ -2561,8 +2650,14 @@ int32 unit_skilluse_pos2( struct block_list *src, int16 skill_x, int16 skill_y,
 #endif
 
 	ud->state.skillcastcancel = castcancel&&casttime>0?1:0;
-	if (!sd || sd->skillitem != skill_id || skill_get_cast(skill_id, skill_lv))
-		ud->canact_tick = tick + i64max(casttime, max(status_get_amotion(src), battle_config.min_skill_delay_limit)) + SECURITY_CASTTIME;
+
+	// Skills used from items don't seem to give any attack or act delay
+	if (sd == nullptr || sd->skillitem != skill_id) {
```

Comment:
I think this logic should be moved into unit_set_attackdelay.
The function call should be moved after setting ud->skill_id, then you access it in unit_set_attackdelay

Reply:
See above.